pkg/cli/interactive_tests/test_connect_cmd.tcl

```diff
@@ -0,0 +1,101 @@
+#! /usr/bin/env expect -f
+
+source [file join [file dirname $argv0] common.tcl]
+
+set certs_dir "/certs"
+set ::env(COCKROACH_INSECURE) "false"
+set ::env(COCKROACH_HOST) "localhost"
+
+proc start_secure_server {argv certs_dir extra} {
+    report "BEGIN START SECURE SERVER"
+    system "$argv start-single-node --host=localhost --socket-dir=. --certs-dir=$certs_dir --pid-file=server_pid -s=path=logs/db --background $extra >>expect-cmd.log 2>&1;
+            $argv sql --certs-dir=$certs_dir -e 'select 1'"
+    report "END START SECURE SERVER"
+}
+
+proc stop_secure_server {argv certs_dir} {
+    report "BEGIN STOP SECURE SERVER"
+    system "$argv quit --certs-dir=$certs_dir"
+    report "END STOP SECURE SERVER"
+}
+
+start_secure_server $argv $certs_dir ""
+
+spawn $argv sql --certs-dir=$certs_dir
+eexpect root@
+
+start_test "Test initialization"
+send "create database t; set database = t;\r"
+eexpect root@
+eexpect "/t>"
+
+start_test "Check that the client-side connect cmd prints the current conn details"
+send "\\c\r"
+eexpect "Connection string:"
+eexpect "You are connected to database \"t\" as user \"root\""
+eexpect root@
+eexpect "/t>"
+
+start_test "Check that the client-side connect cmd changes databases using a SET statement"
+send "\\c system\r"
+eexpect SET
+eexpect root@
+eexpect "/system>"
+end_test
+
+start_test "Check that the client-side connect cmd can change users using a password"
+send "create user foo with password 'abc';\r"
+eexpect "CREATE ROLE"
+eexpect root@
+
+send "\\c - foo\r"
+eexpect "using new connection URL"
+eexpect "Connecting to server"
+eexpect "as user \"foo\""
+eexpect "Enter password:"
+send "abc\r"
+eexpect foo@
+eexpect "/system>"
+end_test
+
+start_test "Check that the client-side connect cmd can change hosts"
+send "\\c - - localhost\r"
+eexpect "using new connection URL"
+eexpect "Connecting to server"
+eexpect "as user \"foo\""
+eexpect "Enter password:"
+send "abc\r"
+eexpect foo@
+eexpect "/system>"
+end_test
+
+start_test "Check that the client-side connect cmd can change ports"
+send "\\c - - - 26257\r"
+eexpect "using new connection URL"
+eexpect "Connecting to server"
+eexpect "as user \"foo\""
+eexpect "Enter password:"
+send "abc\r"
+eexpect foo@
+eexpect "/system>"
+end_test
+
+start_test "Check that the client-side connect cmd can detect syntax errors"
+send "\\c - - - - abc\r"
+eexpect "unknown syntax"
+eexpect foo@
+eexpect "/system>"
+end_test
+
+start_test "Check that the client-side connect cmd can change users with certs using a URL"
+send "\\c postgres://root@localhost:26257/defaultdb?sslmode=require&sslcert=$certs_dir%2Fclient.root.crt&sslkey=$certs_dir%2Fclient.root.key&sslrootcert=$certs_dir%2Fca.crt\r"
+eexpect "using new connection URL"
+eexpect root@
+eexpect "/defaultdb>"
+end_test
```

Comment:
can you add tests that checks whether this exits cleanly if the new connection points to an invalid URL / db?